src/components/purchase-progress.js

```diff
@@ -40,12 +41,13 @@ class PurchaseProgress extends Component {
     const { currentStep, maxStep, perspective, purchase, subdued } = this.props
     const { progressCalculated, progressWidth } = this.state
 
-    // timestamps not yet available
-    const soldAt = !!currentStep
-    const fulfilledAt = currentStep > 1
-    const receivedAt = currentStep > 2
-    const withdrawnAt = currentStep > 3
-    const disputed = purchase && purchase.status === 'disputed'
+    const offerCreated = purchase && purchase.event('OfferCreated')
+    const offerWithdrawn = purchase && purchase.event('OfferWithdrawn')
+    const offerAccepted = purchase && purchase.event('OfferAccepted')
+    const offerDisputed = purchase && purchase.event('OfferDisputed')
+    const offerRuling = purchase && purchase.event('OfferRuling')
+    const offerFinalized = purchase && purchase.event('OfferFinalized')
+    const offerData = purchase && purchase.event('OfferData')
```

Comment:
These are repeated across components as well... could it be a util that returns an object with all of these values?